libfs/file.go

```diff
@@ -218,14 +218,34 @@ func (f *File) Unlock() (err error) {
 		return f.fs.config.MDServer().ReleaseLock(f.fs.ctx,
 			f.fs.root.GetFolderBranch().Tlf, f.getLockID())
 	}
-	return jServer.FinishSingleOp(f.fs.ctx,
+	err = jServer.FinishSingleOp(f.fs.ctx,
 		f.fs.root.GetFolderBranch().Tlf, &keybase1.LockContext{
 			RequireLockID:       f.getLockID(),
 			ReleaseAfterSuccess: true,
 		}, f.fs.priority)
+	if err != nil {
+		return err
+	}
+
+	if f.fs.config.Mode() != libkbfs.InitSingleOp {
+		f.fs.log.CDebugf(f.fs.ctx, "Releasing the lock")
+
+		// Need to explicitly release the lock from the server.
```

Comment:
why?

Reply:
Because if single-op mode isn't enabled, then the journal will be flushing on its own without waiting for the call to `FinishSingleOp`.  That means the journal can already be completely flushed by the time `FinishSingleOp` is called, and it will be a no-op.  It won't have made any call to the server to release the lock, so we have to do it explicitly here.

Perhaps there's a way to structure `FinishSingleOp` so that it can know whether the lock was released as part of the journal flush or not, and it can release the lock explicitly if so.  I don't think that will be trivial, but I can look into it if you all want.  Or I can just expand the comment here.  What do you think?

Reply:
I'm fine with an expanded comment (which can just be your first paragraph here).